rita_client/src/exit_manager/mod.rs

```diff
@@ -72,78 +76,121 @@ pub struct ExitBlacklist {
+    pub nat_setup: bool,
     /// Every tick we query an exit endpoint to get a list of exits in that cluster. We use this list for exit switching
     pub exit_list: ExitList,
+    /// Store last exit here, when we see an exit change, we reset wg tunnels
+    pub last_exit: Option<IpAddr>,
+}
+
+pub fn set_last_exit(e: Option<IpAddr>, em_state: &mut ExitManager) {
+    em_state.last_exit = e;
+}
+
+pub fn get_last_exit(em_state: ExitManager) -> Option<IpAddr> {
+    em_state.last_exit
 }
 
 /// This functions sets the exit list ONLY IF the list arguments provived is not empty. This is need for the following edge case:
 /// When an exit goes down, the endpoint wont repsond, so we have no exits to switch to. By setting only when we have a length > 1
 /// we assure that we switch when an exit goes down
-pub fn set_exit_list(list: ExitList) -> bool {
+pub fn set_exit_list(list: ExitList, em_state: &mut ExitManager) -> bool {
     if !list.exit_list.is_empty() {
-        EXIT_MANAGER.write().unwrap().exit_list = list;
+        em_state.exit_list = list;
         return true;
     }
     false
 }
 
-pub fn get_exit_list() -> ExitList {
-    EXIT_MANAGER.read().unwrap().exit_list.clone()
+pub fn get_exit_list(em_state: ExitManager) -> ExitList {
+    em_state.exit_list
 }
 
 pub fn get_selected_exit(exit: String) -> Option<IpAddr> {
-    match EXIT_MANAGER.read().unwrap().selected_exit_list.get(&exit) {
+    match SELECTED_EXIT_LIST
+        .write()
+        .unwrap()
+        .selected_exit_list
+        .get(&exit)
+    {
         Some(a) => a.selected_id,
         None => None,
     }
 }
 
 pub fn get_selected_exit_metric(exit: String) -> Option<u16> {
-    match EXIT_MANAGER.read().unwrap().selected_exit_list.get(&exit) {
+    match SELECTED_EXIT_LIST
+        .read()
+        .unwrap()
+        .selected_exit_list
+        .get(&exit)
+    {
         Some(a) => a.selected_id_metric,
         None => None,
     }
 }
 
 pub fn get_selected_exit_tracking(exit: String) -> Option<IpAddr> {
-    match EXIT_MANAGER.read().unwrap().selected_exit_list.get(&exit) {
+    match SELECTED_EXIT_LIST
+        .read()
+        .unwrap()
+        .selected_exit_list
+        .get(&exit)
+    {
         Some(a) => a.tracking_exit,
         None => None,
     }
 }
 
 pub fn get_selected_exit_degradation(exit: String) -> Option<u16> {
-    match EXIT_MANAGER.read().unwrap().selected_exit_list.get(&exit) {
+    match SELECTED_EXIT_LIST
+        .read()
+        .unwrap()
+        .selected_exit_list
+        .get(&exit)
+    {
         Some(a) => a.selected_id_degradation,
         None => None,
     }
 }
 
 pub fn set_selected_exit(exit: String, exit_info: SelectedExit) {
-    EXIT_MANAGER
+    SELECTED_EXIT_LIST
         .write()
         .unwrap()
         .selected_exit_list
         .insert(exit, exit_info);
 }
 
-pub fn set_em_nat(val: bool) {
-    EXIT_MANAGER.write().unwrap().nat_setup = val;
+pub fn set_em_nat(val: bool, em_state: &mut ExitManager) {
```

Comment:
I don't see what this function helps with?

Reply:
removed